File: src/06.hashMap/14.167.go
package hashmap

func TwoSum1(nums []int, target int) []int {
	hashMap := make(map[int]int)

	for i, v := range nums {
		diff := target - v
		if val, ok := hashMap[diff]; ok {
			return []int{val, i}
		}
		hashMap[v] = i
	}

	return []int{}
}

File: src/06.hashMap/14.167.ts
export function twoSum1(nums: number[], target: number): number[] {
    // for (let i = 0; i < nums.length - 1; i++) {
    //     for (let j = i + 1; j < nums.length; j++) {
    //         if (nums[i] + nums[j] === target) {
    //             return [i, j];
    //         }
    //     }
    // }
    // return [];

    const hashMap = new Map<number, number>();
    for (let i = 0; i < nums.length; i++) {
        const diff = target - nums[i];
        if (hashMap.has(diff)) {
            return [hashMap.get(diff), i];
        }

        hashMap.set(nums[i], i);
    }
    return [];
}
